### DISCUSSION TITLE
knee_singlecoil content was inconsistent with the manifest

### DISCUSSION THREAD

**Discussion by Liuzheng0808** [facebookresearch/fastMRI#295] [Q&A] [answered]
Hello, I've been trying to use knee_singlecoil recently,but when I downloaded it, I found that the content was inconsistent with the manifest.Do you know what the reason is?The downloaded file contains only one file ’knee_singlecoil_train‘ after extraction.

**Comment by mmuckley** [comment]
Okay, this seems to confirm an issue with the single coil test data. I'll track down where the issue is.

**Reply by mmuckley** [reply]
@Liuzheng0808 could you try re-downloading the data now?

**Reply by Liuzheng0808** [reply]
Hello, I have tried to re-download the knee_singlecoil_val,but there is still only one file in it called ’knee_single_val'.It's not what the list of the manifest shows.

**Reply by Liuzheng0808** [reply]
Is it a question of downloading links?

**Reply by Liuzheng0808** [reply]
@mmuckley Hello, I was wondering if I need to further process the downloaded file to produce the file in the list ? Thank you for your help.

**Reply by mmuckley** [reply]
Hello @Liuzheng0808, I thought you were talking about the test set.

Based on what I can see with current download links, the validation set should be pretty big and should definitely have more than one file. Perhaps there was a bug in your curl command for downloading?

**Reply by Liuzheng0808** [reply]
Sorry, I'm sure I'm downloading the training and validation sets. In the manifest, they have more than one file, such as file10001.h5,file10002.h5.......However, in the files I downloaded, they only had one file. In the train set，it called ‘knee_singlecoil_train.h5';  In the val set, it called 'knee_singlecoil_val.h5'. Moreover, the space they occupy has not changed，about 80G and 14G. When I read them separately, they show the following：
{
val
ismrmrd_header <HDF5 dataset "ismrmrd_header": shape (), type "|O">
kspace <HDF5 dataset "kspace": shape (36, 640, 368), type "<c8">
reconstruction_esc <HDF5 dataset "reconstruction_esc": shape (36, 320, 320), type "<f4">
reconstruction_rss <HDF5 dataset "reconstruction_rss": shape (36, 320, 320), type "<f4">

train
ismrmrd_header <HDF5 dataset "ismrmrd_header": shape (), type "|O">
kspace <HDF5 dataset "kspace": shape (33, 640, 368), type "<c8">
reconstruction_esc <HDF5 dataset "reconstruction_esc": shape (33, 320, 320), type "<f4">
reconstruction_rss <HDF5 dataset "reconstruction_rss": shape (33, 320, 320), type "<f4">
}
They look like the contents of one file（for example file10001.h5).
I don't know why this is the case, do you have a way to solve it？
Or there was a bug in my curl command for downloading？

Thank you again for your patience and help.

**Reply by Liuzheng0808** [reply]
@mmuckley

**Reply by mmuckley** [reply, marked as answer]
@Liuzheng0808

I just downloaded it myself and extracted the files to verify that they match the manifest for `knee_singlecoil_val_tar.xz`, so I think you've made an error.

You need to do the following:

1. Run the full `curl` command from your email. The last two items in the command should be `--output knee_singlecoil_val.tar.xz`.
2. Run `tar -xvf knee_singlecoil_val.tar.xz` on the downloaded file, which should be named `knee_singlecoil_val.tar.xz`.

I'm not sure why your `.h5` files are the correct sizes. My initial guess is that you renamed the `knee_single_coil_val.tar.xz` file to `knee_singlecoil_val.h5` and your HDF5 script has somehow found the first volume in the file, but to be honest I can't be sure. If that were the case then you could be able to rename `knee_singlecoil_val.h5` to `knee_single_coil_val.tar.xz` and run `tar -xvf` on that.

**Reply by Liuzheng0808** [reply]
@mmuckley Thank you so much! I successfully extracted the files！Like you said, I rename knee_singlecoil_val.h5 to knee_singlecoil_val.tar.xz.
It worked! Thank you again for your help！I hope you have a good life！
